daemon/list_test.go

```diff
@@ -0,0 +1,62 @@
+package daemon
+
+import (
+	"testing"
+
+	"github.com/docker/docker/api/types/filters"
+)
+
+func TestFilterEvents(t *testing.T) {
+	containerFilter := filters.NewArgs()
+	containerFilter.Add("container", "dasf3423dags")
+
+	daemonFilter := filters.NewArgs()
+	daemonFilter.Add("daemon", "adsf324")
+
+	eventFilter := filters.NewArgs()
+	eventFilter.Add("event", "die")
+
+	imageFilter := filters.NewArgs()
+	imageFilter.Add("image", "sadsfwewre")
+
+	labelFilter := filters.NewArgs()
+	labelFilter.Add("label", "a=b")
+	labelFilter.Add("label", "c")
+
+	networkFilter := filters.NewArgs()
+	networkFilter.Add("network", "bridge")
+
+	pluginFilter := filters.NewArgs()
+	pluginFilter.Add("plugin", "bridge")
+
+	typeFilter := filters.NewArgs()
+	typeFilter.Add("type", "container")
+
+	volumeFilter := filters.NewArgs()
+	volumeFilter.Add("volume", "asdwqfs34s")
+
+	validEventFilters := []filters.Args{
+		containerFilter,
+		daemonFilter,
+		eventFilter,
+		imageFilter,
+		labelFilter,
+		networkFilter,
+		pluginFilter,
+		typeFilter,
+		volumeFilter,
+	}
+
+	for _, singleFilter := range validEventFilters {
+		if err := singleFilter.Validate(acceptedEventFilters); err != nil {
+			t.Fatalf("expect no error, got error '%s'", err)
+		}
+	}
+
+	invalidFilter := filters.NewArgs()
+	invalidFilter.Add("nonexisttype", "abcd")
+
+	if err := invalidFilter.Validate(acceptedEventFilters); err == nil {
+		t.Fatalf("expect an error, while got none error")
```

Comment:
Is there an error message we check here, instead of just checking that there is an error?

```go
assert.EqualError(t, err, expectedErr)
```

or

```go
testutil.ErrorContains(t, err, errMessage)
```